src/flask_uploads/flask_uploads.py

```diff
@@ -79,7 +78,7 @@ def config_for_set(
             # use the default dest from the config
             if defaults['dest'] is not None:
                 using_defaults = True
-                destination = os.path.join(defaults['dest'], uset.name)
+                destination = str(Path(defaults['dest']) / uset.name)
```

Comment:
Working with path objects can be seen differently like working with encodings. In the sphere of your control everything should be a path object. Only at the borders conversions should be done. So if a path comes in (e.g. from the command line, a web form, etc.) it should immediately be transformed into a path object and then it should stay like that until it has to leave the sphere of your control again (e.g. when calling a 3rd party library that cannot cope (yet) with path objects).